vk/router.go

```diff
@@ -139,6 +149,52 @@ func (rt *Router) handlerWrap(inner HandlerFunc) httprouter.Handle {
 	}
 }
 
+// wsHandlerWrap returns an httprouter.Handle that uses the `inner` vk.WebSocketHandleFunc to handle the request
+//
+// inner accepts a Gorilla `Conn` and reads and writes messages to it
+//
+func (rt *Router) wsHandlerWrap(inner WebSocketHandlerFunc) httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+		var status int
+		var body []byte
+
+		// create a context handleWrap the configured logger
+		// (and use the ctx.Log for all remaining logging
+		// in case a scope was set on it)
+		ctx := NewCtx(rt.log, params, w.Header())
+		ctx.UseScope(defaultScope{ctx.RequestID()})
+
+		logDone := rt.logRequest(r, ctx)
+
+		upgrader := websocket.Upgrader{
+			ReadBufferSize:  1024,
+			WriteBufferSize: 1024,
+			CheckOrigin:     func(r *http.Request) bool { return true },
+		}
+
+		conn, err := upgrader.Upgrade(w, r, nil)
+		if err != nil {
+			status, body, _ = errorOrOtherToBytes(ctx.Log, err)
+			w.WriteHeader(status)
+			w.Write(body)
+
+			logDone(status)
+			return
+		}
+
+		err = inner(r, ctx, conn)
+
+		if err != nil {
+			// tbd what we'll do with the error
```

Comment:
Looking for some opinions here as well. As far as I know, there isn't really a "standard" way to report that something went terribly wrong other than just closing the connection.

Reply:
https://www.rfc-editor.org/rfc/rfc6455.html#section-7.2

Reply:
^ this was incorrect 

After the websocket handshake has succeeded I think you have two options. 

1. Abort the session (close the connection)
2. Send back a standard data frame with an error message in the payload (basically write your own builderservice messaging protocol)